### ISSUE TITLE
Self-transform MODS function

### ISSUE THREAD

**Issue by bondjimbond** [SFULibrary/islandora_datastream_crud#75]
A nice new feature would be to update the MODS datastreams for a list of PIDs by running them through a given self-transform XSL.

Use case: I've got a [pull request open for Islandora XML Forms](https://github.com/Islandora/islandora_xml_forms/pull/264) that provides a new self-transform XSLT, which I would like to apply against every object with a MODS datastream in the repository (~176,000 objects).

**Comment by mjordan** [comment]
This would be a useful feature... but maybe we can extend it to allow the use of an arbitrary XSLT stylesheet. If the user wanted to use the self transform configured in XML forms, they would pass a specific value to the drush command, but if they wanted to use an external XSLT stylesheet, they could pass the path to it.

**Comment by mjordan** [comment]
Of course, we don't want to make it specific to MODS, we'd need to allow the user to indicate the DSID that would be updated.

**Comment by bondjimbond** [comment]
Sounds good to me!

**Comment by mjordan** [comment]
Over the weekend, I will test and merge (unless there are large issues) #74. Do you want me to take an initial stab at implementing this new feature as well?

**Comment by bondjimbond** [comment]
Sure, no reason not to if you have the time!

**Comment by mjordan** [comment]
Just a note here, but this feature will need to take the behavior described in the "Updating DC datastreams by pushing other XML datastreams" section of the readme into account, which means we'll probably need an option to also update the DC or not.

**Comment by bondjimbond** [comment]
Absolutely! And default behaviour should be to update DC if updating MODS.

**Comment by mjordan** [comment]
I was thinking the default behavior would be to do as it does now - prompt you for a yes or no. But I'll have to think about it a bit more.

**Comment by bondjimbond** [comment]
Yes, that's sensible.
